pennylane_lightning/core/_serialize.py

```diff
@@ -254,23 +257,31 @@ def _pauli_sentence(self, observable, wires_map: dict = None):
         terms = [self._pauli_word(pw, wires_map) for pw in pwords]
         coeffs = np.array(coeffs).astype(self.rtype)
 
+        # TODO: Add this
+        # if len(terms) == 1 and coeffs[0] == 1.0:
+        #     return terms[0]
+
         if self.split_obs:
             return [self.hamiltonian_obs([c], [t]) for (c, t) in zip(coeffs, terms)]
         return self.hamiltonian_obs(coeffs, terms)
 
-    # pylint: disable=protected-access
+    # pylint: disable=protected-access, too-many-return-statements
     def _ob(self, observable, wires_map: dict = None):
         """Serialize a :class:`pennylane.operation.Observable` into an Observable."""
-        if isinstance(observable, Tensor):
+        if isinstance(observable, (Prod, Sum, SProd)) and observable.pauli_rep is not None:
+            return self._pauli_sentence(observable.pauli_rep, wires_map)
+        if isinstance(observable, (Tensor, Prod)):
             return self._tensor_ob(observable, wires_map)
-        if observable.name == "Hamiltonian":
+        if observable.name in ("Hamiltonian", "LinearCombination"):
             return self._hamiltonian(observable, wires_map)
         if observable.name == "SparseHamiltonian":
             return self._sparse_hamiltonian(observable, wires_map)
         if isinstance(observable, (PauliX, PauliY, PauliZ, Identity, Hadamard)):
             return self._named_obs(observable, wires_map)
-        if observable._pauli_rep is not None:
-            return self._pauli_sentence(observable._pauli_rep, wires_map)
+        if observable.pauli_rep is not None:
+            return self._pauli_sentence(observable.pauli_rep, wires_map)
+        # if isinstance(observable, (Prod, Sum)):
+        #     return self._hamiltonian(observable, wires_map)
```

Comment:
Optimization for later

Reply:
Does it mean `Prod` won't be handled by `_tensor_ob`? Or not, because operators are allowed to act on overlapping wires?

Reply:
`Prod` does allow overlapping wires, true. I hadn't considered that, so maybe we check if there are overlapping ops, and if that's the case then we dispatch to the `Hermitian` pipeline, otherwise `Tensor`. The reason I was testing out `Hamiltonian` was because `Prod` can have `Sum` ops inside it, but I think it's better to just send it to the `Tensor` pipeline.